tests/integration-tests/tests/storage/test_fsx_lustre.py

```diff
@@ -304,7 +308,58 @@ def create_fsx_open_zfs(fsx_factory, num):
     volume_list = boto3.client("fsx").describe_volumes(Filters=[{"Name": "file-system-id", "Values": file_system_ids}])[
         "Volumes"
     ]
-    return [(volume["FileSystemId"], volume["VolumeId"]) for volume in volume_list]
+    return [volume["VolumeId"] for volume in volume_list]
+
+
+@pytest.fixture(scope="class")
+def open_zfs_volume_factory(vpc_stack, cfn_stacks_factory, request, region, key_name):
+    """
+    Define a fixture to manage the creation and destruction of volumes for FSx for OpenZFS.
+
+    return volume ids
+    """
+    created_stacks = []
+
+    def _open_zfs_volume_factory(root_volume_id, num_volumes=1):
+        fsx_open_zfs_volume_template = Template()
+        fsx_open_zfs_volume_template.set_version()
+        fsx_open_zfs_volume_template.set_description("Create Storage Virtual Machine stack")
+
+        open_zfs_volume_resource_name = "OpenZFSVolume"
+        max_concurrency = 15
+        for index in range(num_volumes):
+            depends_on_arg = {}
+            if index >= max_concurrency:
+                depends_on_arg = {"DependsOn": [f"{open_zfs_volume_resource_name}{index - max_concurrency}"]}
+            fsx_open_zfs_volume = Volume(
+                title=f"{open_zfs_volume_resource_name}{index}",
+                Name=f"vol{index}",
+                VolumeType="OPENZFS",
+                OpenZFSConfiguration=VolumeOpenZFSConfiguration(
+                    NfsExports=[
+                        NfsExports(ClientConfigurations=[ClientConfigurations(Clients="*", Options=["rw", "crossmnt"])])
+                    ],
+                    ParentVolumeId=root_volume_id,
+                ),
+                **depends_on_arg,
+            )
+            fsx_open_zfs_volume_template.add_resource(fsx_open_zfs_volume)
+        fsx_stack = CfnStack(
+            name=utils.generate_stack_name(
+                "integ-tests-fsx-openzfs-volume", request.config.getoption("stackname_suffix")
+            ),
+            region=region,
+            template=fsx_open_zfs_volume_template.to_json(),
+        )
+        cfn_stacks_factory.create_stack(fsx_stack)
+        created_stacks.append(fsx_stack)
+        return [fsx_stack.cfn_resources[f"{open_zfs_volume_resource_name}{i}"] for i in range(num_volumes)]
+
+    yield _open_zfs_volume_factory
+
+    if not request.config.getoption("no_delete"):
```

Comment:
Providing the possibility to not deleting the stack is really useful.
Nevertheless, isn't it also necessary to provide a way for the test suites to re-execute the tests without re-creating it?

For instance: in the case of the VPC stack, we provide an option to specify the VPC stack name to use for a given tests.

Reply:
Yes. Now we have to change the code manually to reuse the stack. Can we leave this as a further improvement?

Reply:
Sure, just keep track of it.